provider/resource_wireguard_preshared_key.go

```diff
@@ -0,0 +1,55 @@
+package provider
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"golang.zx2c4.com/wireguard/wgctrl/wgtypes"
+	"io"
+)
+
+func resourceWireguardPresharedKey() *schema.Resource {
+	return &schema.Resource{
+		Description: "Provides a WireGuard asymmetric key resource. This can be used to create, read, and delete WireGuard keys in terraform state.",
+
+		Create: resourceWireguardPresharedKeyCreate,
+		Read:   resourceWireguardPresharedKeyRead,
+		Delete: resourceWireguardPresharedKeyDelete,
+
+		Schema: map[string]*schema.Schema{
+			"preshared_key": {
```

Comment:
I think I'd be tempted just to call this `key` - otherwise it's a little verbose to say `wireguard_preshared_key.some-name.preshard_key`?